lxd/storage/backend_lxd.go

```diff
@@ -5198,28 +5221,32 @@ func (b *lxdBackend) CreateCustomVolumeSnapshot(projectName, volName string, new
 		return err
 	}
 
-	revert := revert.New()
-	defer revert.Fail()
-
-	// Validate config and create database entry for new storage volume.
-	// Copy volume config from parent.
-	err = VolumeDBCreate(b, projectName, fullSnapshotName, parentVol.Description, drivers.VolumeTypeCustom, true, parentVol.Config, time.Now().UTC(), newExpiryDate, drivers.ContentType(parentVol.ContentType), false, true)
+	volDBContentType, err := VolumeContentTypeNameToContentType(parentVol.ContentType)
 	if err != nil {
 		return err
 	}
 
-	revert.Add(func() { _ = VolumeDBDelete(b, projectName, fullSnapshotName, drivers.VolumeTypeCustom) })
-
-	volDBContentType, err := VolumeContentTypeNameToContentType(parentVol.ContentType)
+	contentType, err := VolumeDBContentTypeToContentType(volDBContentType)
 	if err != nil {
 		return err
 	}
 
-	contentType, err := VolumeDBContentTypeToContentType(volDBContentType)
+	if contentType != drivers.ContentTypeFS && contentType != drivers.ContentTypeBlock {
+		return fmt.Errorf("Volume of content type %q cannot be snapshot", contentType)
```

Comment:
```suggestion
		return fmt.Errorf("Volume of content type %q cannot be snapshotted", contentType)
```

Reply:
@ru-fu @tomponline I'm not certain on this one. There's another occurrence of this where is says `Volume cannot be snapshot`. Is is `snapshot` or `snapshotted`?

Reply:
Let's go for "Volume of content type %q does not support snapshots" - that sounds much better in any case :)

Reply:
And if anybody's curious - looks like most dictionaries that actually have snapshot as a verb allow both "snapshot" and "snapshotted" as past participle. :)